angular/core/components/thread_page/context_panel/context_panel.coffee

```diff
@@ -0,0 +1,52 @@
+angular.module('loomioApp').directive 'contextPanel', ->
+  scope: {discussion: '='}
+  restrict: 'E'
+  replace: true
+  templateUrl: 'generated/components/thread_page/context_panel/context_panel.html'
+  controller: ($scope, $rootScope, $window, $timeout, AbilityService, Session, ModalService, ChangeVolumeForm, DiscussionForm, ThreadService, MoveThreadForm, PrintModal, DeleteThreadForm, RevisionHistoryModal) ->
+
+    $scope.showContextMenu = ->
+      AbilityService.canChangeThreadVolume($scope.discussion)
+
+    $scope.canChangeVolume = ->
+      Session.user().isMemberOf($scope.discussion.group())
+
+    $scope.openChangeVolumeForm = ->
+      ModalService.open ChangeVolumeForm, model: => $scope.discussion
+
+    $scope.canEditThread = ->
+      AbilityService.canEditThread($scope.discussion)
+
+    $scope.editThread = ->
+      ModalService.open DiscussionForm, discussion: => $scope.discussion
+
+    $scope.muteThread = ->
+      ThreadService.mute($scope.discussion)
+
+    $scope.unmuteThread = ->
+      ThreadService.unmute($scope.discussion)
+
+    $scope.canMoveThread = ->
+      AbilityService.canMoveThread($scope.discussion)
+
+    $scope.moveThread = ->
+      ModalService.open MoveThreadForm, discussion: => $scope.discussion
+
+    $scope.requestPagePrinted = ->
+      unless discussion.allEventsLoaded()
```

Comment:
Hi I'm going to say this but you can ignore it if you like.

I'm not a fan of  the leading 'unless'; it doesn't read well in my head and gets the story of the method backwards. 'If' is optimistic; it says, 'this is what should happen, but if it doesn't we've got a backup'. Which is a much better sentence than 'unless it's sunny today, we will stay inside, otherwise we will go to the park'

So this imo should be
```
if allEventsLoaded
  # print
else
  # load and print
```